### PR TITLE
Add syscall commands

### PR DESCRIPTION
**Pull request by daniellimws**
### Description/Motivation/Screenshots ###
This was mentioned earlier in #257 

This patch allows gdb to retrieve syscall related information from syscall-tables in gef-extras so that the user can quickly check what is the next syscall instruction going to do without needing to refer externally.

Hopefully, this can reduce the hassle of needing to open another tab in the browser to refer to syscalls every time we encounter one.

https://asciinema.org/a/gNwy6khs3rkQAPyv1CMrFAnlf

### How Has This Been Tested? ###
Refer to asciinema above.

| Architecture | Yes/No                   | Comments               |
|--------------|:------------------------:|------------------------|
| x86-32       | :heavy_check_mark:  |                  |
| x86-64       |  :heavy_check_mark:  |                  |
| ARM          | :heavy_multiplication_x: |                        |
| AARCH64      | :heavy_multiplication_x: |                        |
| MIPS         | :heavy_multiplication_x: |                        |
| POWERPC      | :heavy_multiplication_x: |                        |
| SPARC        | :heavy_multiplication_x: |                        |

### Checklist ###

- [x] My code follows the code style of this project.
- [ ] My change includes a change to the documentation, if required.
- [x] I have read and agree to the **CONTRIBUTING** document.
- [ ] I have tagged the PR as appropriate (e.g. bug fix).

### CODE REVIEW

**Comment by daniellimws** [comment]
I will add documentation tomorrow.

File: gef.py [hunk 1]
```diff
@@ -1571,6 +1573,7 @@ class X86(Architecture):
         21: "identification",
     }
     syscall_register = "$eax"
+    syscall_instructions = ['sysenter', 'int 0x80']
```
Comment (Grazfather) [inline gef.py]: nitpick: double quotes

File: gef.py [hunk 2]
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
...
+
+            gef_print(line)
+
+        return
+
+    def get_filepath(self, x):
+        p = self.get_settings_path()
+        if not p: return None
+        return os.path.join(p, "{}.py".format(x))
+
```
Comment (Grazfather) [inline gef.py]: nit: extra line

File: gef.py [hunk 3]
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
...
+        self.add_setting("path", os.path.join(GEF_TEMP_DIR, "syscall-tables"),
+                         "Path to store/load the syscall tables files")
+        return
+
+    def do_invoke(self, argv):
+        color = get_gef_setting("theme.xinfo_title_message")
+
+        path = self.get_settings_path()
+        if path is None:
+            err("Cannot open '{0}': check directory and/or `gef config {0}` setting, currently: '{1}'".format("syscall-args.path",
```
Comment (Grazfather) [inline gef.py]: Long line, let's keep them under 120 chars

File: gef.py [hunk 4]
```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
...
+        parameters = [s.param for s in syscall_entry.params]
+        registers = [s.reg for s in syscall_entry.params]
+
+        info("Detected syscall {}".format(Color.colorify(syscall_entry.name, attrs=color)))
+        gef_print("    {}({})".format(syscall_entry.name, ', '.join(parameters)));
+
+        headers = [Color.colorify(x, attrs=color) for x in ["Parameter", "Register", "Value"]]
+        param_names = [re.split(' |\*', p)[-1] for p in parameters]
+        info("{:<28} {:<28} {}".format(*headers))
+        for i in range(len(parameters)):
```
Comment (Grazfather) [inline gef.py]: I think I'd prefer `for name, register, value in zip(param_names, registers, values):` or similar.

**Comment by daniellimws** [comment]
Alright fixed those issues

**Comment by daniellimws** [comment]
Updated docs as well.
Somehow the descriptions don't show up next to the list of commands when I enter `gef` in gdb. I've added comments at the top of the class of each command, am I doing it right?

**Comment by Grazfather** [comment]
The issue with the command listing is that you have a newline at the start of your docstring. This is a bug in gef. Filed #285.

**Pull request merged**